### PR TITLE
Add fsync/chmod/realpath/chdir/getcwd/fdopendir/etc implementations 

### LINKED ISSUE #8: stat returns EINVAL
`stat` returns EINVAL for an empty path, but it [should return ENOENT](http://pubs.opengroup.org/onlinepubs/9699919799/functions/fstatat.html).

### PR DESCRIPTION
**Pull request by isage**
* add getcwd/chdir implementations.
* Rewrite open/mkdir/etc. to use cwd.
* Reformat code to follow one coding standard (we can discuss and reformat that later, for now i choose one that was used mostly through code).
* Rewrite error handling.
* rewrite dirent to use open/close and vita fdmap.
* add fdopendir/scandir/dirfd/alphasort.
* add fsync
* add support for opening directories with `open`
* fix getentropy to allow 256 buffer

This effectively makes #63 and #64 obsolete. Also fixes #8 due to path checking and #36.
I did some basic tests and everything seems to work, but it still requires thorough review and testing.

### CODE REVIEW

**Review by v-atamanenko — APPROVED** [review]
Read the code, tested on my projects (@isage already implemented the comments I had), works good.

File: newlib/libc/sys/vita/dirent.c [hunk 1]
```diff
@@ -24,74 +24,93 @@ DEALINGS IN THE SOFTWARE.
...
-		errno = uid & SCE_ERRNO_MASK;
-		return NULL;
-	}
-
 	DIR *dirp = calloc(1, sizeof(DIR));
 
 	if (!dirp)
 	{
-		sceIoDclose(uid);
+		close(fd);
```
Comment (d3m3vilurr) [inline newlib/libc/sys/vita/dirent.c]: is this right?
when call it from `fdopendir`, this chain would close the fd in here.

how about change the opendir to;
```
DIR *opendir(const char *dirname)
{
  int fd;
  if ((fd = ...) == -1)
    return NULL;
  DIR *ret = __opendir_common(fd);
  if (!ret) {
    close(fd);
  }
  return ret;
}
```

**Review by d3m3vilurr — CHANGES_REQUESTED** [review]

File: newlib/libc/sys/vita/dirent.c (hunk 1 above)
Comment (isage) [reply newlib/libc/sys/vita/dirent.c]: Yeah. Nice catch.

File: newlib/libc/sys/vita/fs.c [hunk 2]
```diff
@@ -43,10 +49,480 @@ int mkdir(const char *pathname, mode_t mode)
...
+	return _stat_r(_REENT, path, buf);
+}
+
+int chmod(const char *pathname, mode_t mode)
+{
+	struct SceIoStat stat = {0};
+	char* full_path = __realpath(pathname);
+	if(!full_path)
+	{
+		errno = errno; // set by realpath
```
Comment (d3m3vilurr) [inline newlib/libc/sys/vita/fs.c]: ENOENT or EIO?

Comment (isage) [reply newlib/libc/sys/vita/fs.c]: EINVAL/ENOENT/ENOMEM.Removed redundant assignments from rmdir and chmod, leaving only comment.

**Pull request merged**